### PR TITLE
Support Array type

### LINKED ISSUE #16301: [Mobile] I get an error that nothing but strings are allowed, even though they are strings.
### Describe the issue

I get an error that nothing but strings are allowed, even though they are strings.

`import {Env, Tensor} from "onnxruntime-react-native";
import * as ort from "onnxruntime-react-native";
import RNFS from "react-native-fs";
import {Image} from 'react-native';
export async function post(message){
    const dest = RNFS.DocumentDirectoryPath+"/monokamo.ort";
    const sess = await ort.InferenceSession.create("file://"+dest)
    const input = new Tensor("string",[message],[1,1])
    const result = await sess.run({text: input});
    console.log(result)
}`

### To reproduce

Run the code pasted above

### Urgency

_No response_

### Platform

React Native

### OS Version

Pixel_3a_API_33_arm64-v8a(AVD)

### ONNX Runtime Installation

Released Package

### Compiler Version (if 'Built from Source')

_No response_

### Package Name (if 'Released Package')

None

### ONNX Runtime Version or Commit ID

1.15.0

### ONNX Runtime API

JavaScript

### Architecture

X64

### Execution Provider

Default CPU

### Execution Provider Library Version

_No response_

### PR DESCRIPTION
**Pull request by mono0218**
Closes #16301 
### Description

Fixed an issue with [this issue](https://github.com/microsoft/onnxruntime/issues/16301)

### CODE REVIEW

**Comment by mono0218** [comment]
@microsoft-github-policy-service agree

**Comment by mono0218** [comment]
> @mono0218 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by mono0218** [comment]
> @mono0218 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by mono0218** [comment]
@microsoft-github-policy-service agree

File: js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java [hunk 1]
```diff
@@ -255,16 +255,18 @@ public WritableMap run(String key, ReadableMap input, ReadableArray output, Read
...
-          //
-          // tensor data should always be a BASE64 encoded string.
-          // This is because the current React Native bridge supports limited data type as arguments.
-          // In order to pass data from JS to Java, we have to encode them into string.
-          //
-          // see also:
-          //   https://reactnative.dev/docs/native-modules-android#argument-types
-          throw new Exception("Non string type of a tensor data is not allowed");
+        if (inputMap.getType("data") != ReadableType.String) {
+          if(inputMap.getType("data") != ReadableType.Array){
```
Comment (fs-eire) [inline js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java:259]: I think `if (inputMap.getType("data") != ReadableType.String && inputMap.getType("data") != ReadableType.Array)` should work?

Comment (mono0218) [reply js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java]: Fixed a simple mistake.

**Review by fs-eire — APPROVED** [review]

**Comment by fs-eire** [comment]
seems there are code conflicts. please merge latest main branch

**Comment by ytaous** [comment]
Please resolve the conflicts, I'll kick off CIs after.

**Pull request closed without merging**